### ISSUE TITLE
When pretrain VO model, loss reduce a few epochs, but immediately increase and remain.

### ISSUE THREAD

**Issue by FL990** [lordmartian/deep_avsr#43]
Hi, thank you for the work.
I reproduced the code, it works well with your pretrained weights.(pretrain, train, test.py all ok)
But I'd like to try some different methods, so I didn't use it this time.
(haven't change the code, just didn't load the pretrained weights)

I was running the **video_only pretrain.py**
Number of Words = 2
My seed = 18547840

And the loss value looks tricky:

Step: 000 || Tr.Loss: 3.135768  Val.Loss: 2.934046 || Tr.CER: 0.977  Val.CER: 0.967 || Tr.WER: 1.022  Val.WER: 1.001
Step: 001 || Tr.Loss: 2.911791  Val.Loss: 2.872616 || Tr.CER: 0.928  Val.CER: 0.890 || Tr.WER: 1.013  Val.WER: 1.000
Step: 002 || Tr.Loss: 2.946833  Val.Loss: 3.349309 || Tr.CER: 0.917  Val.CER: 1.000 || Tr.WER: 1.012  Val.WER: 1.000
Step: 003 || Tr.Loss: 3.326881  Val.Loss: 3.314617 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 004 || Tr.Loss: 3.323197  Val.Loss: 3.303168 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 005 || Tr.Loss: 3.323742  Val.Loss: 3.327703 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 006 || Tr.Loss: 3.324991  Val.Loss: 3.318675 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 007 || Tr.Loss: 3.321263  Val.Loss: 3.317882 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 008 || Tr.Loss: 3.319875  Val.Loss: 3.322166 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 009 || Tr.Loss: 3.323132  Val.Loss: 3.320766 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 010 || Tr.Loss: 3.325111  Val.Loss: 3.319928 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 011 || Tr.Loss: 3.323602  Val.Loss: 3.306033 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 012 || Tr.Loss: 3.322382  Val.Loss: 3.324228 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 013 || Tr.Loss: 3.319179  Val.Loss: 3.308431 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 014 || Tr.Loss: 3.324116  Val.Loss: 3.316021 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000
Step: 015 || Tr.Loss: 3.320856  Val.Loss: 3.307545 || Tr.CER: 1.000  Val.CER: 1.000 || Tr.WER: 1.000  Val.WER: 1.000

On step 1 and 2, the loss reduce to 2.9, but it goes back to 3.3 right next step and remains (it continue until step 130, then I stop it)
And the WER remains 1.00, too.
I found this situation a little different from #6 
Wondering if this give you a clue to know how it happens, or how to avoid this issue?
Or if it's just a normal situation?

Thank you for your kindness.

**Comment by lordmartian** [comment]
Three things come to my mind which you can try:
1. Complete the training for num_words=2 even if the losses don't reduce. Continue the training for higher number of words and see if the losses reduce.
2. Try many vastly varying seed values.
3. Reduce the learning rate and see if that helps.

I am not sure if any of them will work, but you can give it a try.

**Comment by FL990** [comment]
Thank you for these suggestions.
I have tried all of them, and reducing the learning rate seems work for me!
And I'm trying to find a perfect learning rate, if there are any progress, I'll let you know.
Thank you again!
